src/frontend/Sider/Behandling/Vilkårvurdering/utils.ts

```diff
@@ -1,119 +1,30 @@
-    );
-};
-
-export function hentSvaralternativ(
-    regler: Regler,
-    vurdering: Vurdering
-): Svaralternativ | undefined {
-    if (!vurdering.svar) {
-        return undefined;
-    } else {
-        const regel = regler[vurdering.regelId];
-        return regel.svarMapping[vurdering.svar];
-    }
-}
-
-export function leggTilNesteIdHvis(
-    nesteRegelId: RegelId,
-    nySvarArray: Vurdering[],
-    hvisFunksjon: () => boolean
-): Vurdering[] {
-    const inneholderAlleredeNesteRegelId = nySvarArray.some((v) => v.regelId === nesteRegelId);
-    if (nesteRegelId !== 'SLUTT_NODE' && !inneholderAlleredeNesteRegelId && hvisFunksjon()) {
-        return [...nySvarArray, { regelId: nesteRegelId }];
+import { Delvilkår, RegelId, Delvilkårsett } from '../vilkår';
+
+export const finnAvhengighetTilOverordnetValg = (
+    vilkårsvurdering: Delvilkårsett,
+    regelenSomSkalSjekkes: RegelId
+): { følgerAvOverordnetValg: boolean; overordnetValgErOppfylt: boolean | undefined } => {
+    const følgerFraOverordnetValg = vilkårsvurdering[regelenSomSkalSjekkes].følgerFraOverordnetValg;
+
+    if (følgerFraOverordnetValg === null) {
+        return {
+            følgerAvOverordnetValg: false,
+            overordnetValgErOppfylt: undefined,
+        };
     }
-    return nySvarArray;
-}
-
-export const kanHaBegrunnelse = (svarsalternativ: Svaralternativ) =>
-    svarsalternativ.begrunnelseType !== BegrunnelseRegel.UTEN;
 
-export const oppdaterSvarIListe = (
-    nyttSvar: Vurdering,
-    vurderinger: Vurdering[],
-    beholdResterendeSvar = false,
-    beholdBeskrivelse = false
-): Vurdering[] => {
-    const { svar, regelId } = nyttSvar;
+    const { regel, svar } = følgerFraOverordnetValg;
 
-    const svarIndex = vurderinger.findIndex((s) => s.regelId === regelId);
-    const nySvarArray = vurderinger.slice(0, svarIndex);
-
-    if (!svar) {
-        return vurderinger;
-    }
-
-    const oppdatertNyttSvar = beholdBeskrivelse
-        ? { ...nyttSvar, begrunnelse: vurderinger[svarIndex].begrunnelse }
-        : nyttSvar;
-    return [...nySvarArray, oppdatertNyttSvar].concat(
-        beholdResterendeSvar ? [...vurderinger.slice(svarIndex + 1)] : []
-    );
+    return {
+        følgerAvOverordnetValg: true,
+        overordnetValgErOppfylt: vilkårsvurdering[regel].svar === svar,
+    };
 };
 
-const harBegrunnelsePåNesteRegel = (svarsalternativer: Svaralternativ, regler: Regler) =>
-    svarsalternativer.regelId !== 'SLUTT_NODE' &&
-    forventerBegrunnelsePåAlleSvarsalternativ(regler[svarsalternativer.regelId]);
-
-export const kopierBegrunnelse = (
-    tidligereVurderinger: Vurdering[],
-    nyeVurderinger: Vurdering[],
-    nyttSvar: Vurdering,
-    svarsalternativer: Svaralternativ,
-    regler: Regler
-) => {
-    const svarIndex = tidligereVurderinger.findIndex((s) => s.regelId === nyttSvar.regelId);
+export const delvilkårSomErRelevante = (delvilkårsett: Delvilkårsett): [RegelId, Delvilkår][] =>
```

Comment:
Syns denne burde endre navn. Skjønner ikke hva den egentlig gjør og hva "relevante" betyr. Om det betyr å finne de som henger sammen så syntes jeg at endringen av hva et "delvilkår" er ikke er helt optimalt. 

Tidligere bestod delvilkår av flere vurderinger. Hvor alle "følger av annet valg" lå i samme liste, men ting som var uavhengig av hverandre lå hver for seg. Har ikke sett ordentlig på backend enda, så må tenke litt der